SU2_CFD/src/fluid/CSU2TCLib.cpp

```diff
@@ -1829,174 +1927,83 @@ void CSU2TCLib::DiffusionCoeffGY(){
   }
 
   /*--- Mixture thermal conductivity via Gupta-Yos approximation ---*/
-  for (iSpecies = 0; iSpecies < nHeavy; iSpecies++) {
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
 
     /*--- Initialize the species diffusion coefficient ---*/
     DiffusionCoeff[iSpecies] = 0.0;
 
     /*--- Calculate molar concentration ---*/
-    su2double Mi    = MolarMass[iSpecies];
-    su2double gam_i = rhos[iSpecies] / (Density*Mi);
+    const su2double Mi    = MolarMass[iSpecies];
+    const su2double gam_i = rhos[iSpecies] / (Density*Mi);
     su2double denom = 0.0;
-    for (jSpecies = 0; jSpecies < nHeavy; jSpecies++) {
-      if (jSpecies != iSpecies) {
-
-        su2double Mj    = MolarMass[jSpecies];
-        su2double gam_j = rhos[iSpecies] / (Density*Mj);
 
-        /*--- Calculate the Omega^(0,0)_ij collision cross section ---*/
-        su2double Omega_ij = 1E-20 * Omega00(iSpecies,jSpecies,3)
-                            * pow(T, Omega00(iSpecies,jSpecies,0)*log(T)*log(T)
-                                   + Omega00(iSpecies,jSpecies,1)*log(T)
-                                   + Omega00(iSpecies,jSpecies,2));
-        /*--- Calculate "delta1_ij" ---*/
-        su2double d1_ij = 8.0/3.0 * sqrt((2.0*Mi*Mj) / (pi*Ru*T*(Mi+Mj))) * Omega_ij;
-
-        /*--- Calculate heavy-particle binary diffusion coefficient ---*/
-        su2double D_ij = kb*T/(Pressure*d1_ij);
-        denom += gam_j/D_ij;
+    for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
+      if (jSpecies != iSpecies) { 
+
+        const su2double Mj    = MolarMass[jSpecies];
+        const su2double gam_j = rhos[iSpecies] / (Density*Mj);
+
+        const su2double pi = PI_NUMBER;
+        const su2double kb = BOLTZMANN_CONSTANT;
+
+        if (iSpecies == 0) {
+          const su2double d1_ij = ComputeDelta(iSpecies, jSpecies, Mi, Mj, Tve, true);
+          const su2double D_ij = kb*Tve/(Pressure*d1_ij);
+          denom += gam_j/D_ij;
+        } else {
+          const su2double d1_ij = ComputeDelta(iSpecies, jSpecies, Mi, Mj, T, true);
```

Comment:
Good use of these functions.  Maybe we could be more specific with the name "ComputeCollisionalDeltas"?   Just brain storming.

Reply:
Sure, yes that sounds good!